Stripe/Payments/STPThreeDSUICustomization.m

```diff
@@ -25,15 +25,39 @@ + (instancetype)defaultSettings {
         _buttonCustomizationDictionary = [@{
-                                            @(STPThreeDSCustomizationButtonTypeNext): [STPThreeDSButtonCustomization defaultSettingsForButtonType:STPThreeDSCustomizationButtonTypeNext
-                                                                                   ],
-                                            @(STPThreeDSCustomizationButtonTypeCancel): [STPThreeDSButtonCustomization defaultSettingsForButtonType:STPThreeDSCustomizationButtonTypeCancel],
-                                            @(STPThreeDSCustomizationButtonTypeResend): [STPThreeDSButtonCustomization defaultSettingsForButtonType:STPThreeDSCustomizationButtonTypeResend],
-                                            @(STPThreeDSCustomizationButtonTypeSubmit): [STPThreeDSButtonCustomization defaultSettingsForButtonType:STPThreeDSCustomizationButtonTypeSubmit],
-                                            @(STPThreeDSCustomizationButtonTypeContinue): [STPThreeDSButtonCustomization defaultSettingsForButtonType:STPThreeDSCustomizationButtonTypeContinue],
-                                            } mutableCopy];
+                                           @(STPThreeDSCustomizationButtonTypeNext): nextButton,
+                                           @(STPThreeDSCustomizationButtonTypeCancel): cancelButton,
+                                           @(STPThreeDSCustomizationButtonTypeResend): resendButton,
+                                           @(STPThreeDSCustomizationButtonTypeSubmit): submitButton,
+                                           @(STPThreeDSCustomizationButtonTypeContinue): continueButton,
+                                           } mutableCopy];
+        
+        // Initialize the underlying STDS class we are wrapping
+        _uiCustomization = [STDSUICustomization new];
+        [_uiCustomization setButtonCustomization:nextButton.buttonCustomization forType:STDSUICustomizationButtonTypeNext];
+        [_uiCustomization setButtonCustomization:cancelButton.buttonCustomization forType:STDSUICustomizationButtonTypeCancel];
+        [_uiCustomization setButtonCustomization:resendButton.buttonCustomization forType:STDSUICustomizationButtonTypeResend];
+        [_uiCustomization setButtonCustomization:submitButton.buttonCustomization forType:STDSUICustomizationButtonTypeSubmit];
+        [_uiCustomization setButtonCustomization:continueButton.buttonCustomization forType:STDSUICustomizationButtonTypeContinue];
+        
+        _uiCustomization.footerCustomization = _footerCustomization.footerCustomization;
+        _uiCustomization.labelCustomization = _labelCustomization.labelCustomization;
+        _uiCustomization.navigationBarCustomization = _navigationBarCustomization.navigationBarCustomization;
+        _uiCustomization.selectionCustomization = _selectionCustomization.selectionCustomization;
+        _uiCustomization.textFieldCustomization = _textFieldCustomization.textFieldCustomization;
```

Comment:
so the function on line 70 is what was messing me up. Mutating the returned `STPThreeDSButotnCustomization` wasn't syncing. Do the above `_uiCustomization setButtonCustomization` calls fix that because now both dictionaries reference the same STDSButtonCustomization instance?

Reply:
Yep, exactly.  `_buttonCustomizationDictionary ` references `STPThreeDSUIButtonCustomization` instances which forward all their properties to the underlying `STDS` instance, which we are now properly setting on `_uiCustomization` like we should have been (like on line 67, which is why calling the setter synced things).